core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -37,9 +37,60 @@ open class TileInfo {
     var roadStatus = RoadStatus.None
     var turnsToImprovement: Int = 0
 
-    var hasBottomRightRiver = false
-    var hasBottomRiver = false
-    var hasBottomLeftRiver = false
+    fun hasBottomRightRiver(): Boolean {
+        // disable rivers in the map editor for now. otherwise the game crashes.
+        if (!::tileMap.isInitialized) return false
+        tileMap.getRiverBuddies(this.position)
+                .forEach { riverBuddy ->
+                    if (riverBuddy.x == this.position.x - 1 && riverBuddy.y == this.position.y) return true
+                }
+        return false
+    }
+
+    fun hasBottomRiver(): Boolean {
+        if (!::tileMap.isInitialized) return false
+        tileMap.getRiverBuddies(this.position)
+                .forEach { riverBuddy ->
+                    if (riverBuddy.x == this.position.x - 1 && riverBuddy.y == this.position.y - 1) return true
+                }
+        return false
+    }
+
+    fun hasBottomLeftRiver(): Boolean {
+        if (!::tileMap.isInitialized) return false
+        tileMap.getRiverBuddies(this.position)
+                .forEach { riverBuddy ->
+                    if (riverBuddy.x == this.position.x && riverBuddy.y == this.position.y - 1) return true
+                }
+        return false
+    }
+
+    fun hasTopLeftRiver(): Boolean {
+        if (!::tileMap.isInitialized) return false
+        tileMap.getRiverBuddies(this.position)
+                .forEach { riverBuddy ->
+                    if (riverBuddy.x == this.position.x + 1 && riverBuddy.y == this.position.y) return true
+                }
+        return false
+    }
+
+    fun hasTopRiver(): Boolean {
+        if (!::tileMap.isInitialized) return false
+        tileMap.getRiverBuddies(this.position)
+                .forEach { riverBuddy ->
+                    if (riverBuddy.x == this.position.x + 1 && riverBuddy.y == this.position.y + 1) return true
+                }
+        return false
+    }
+
+    fun hasTopRightRiver(): Boolean {
+        if (!::tileMap.isInitialized) return false
+        tileMap.getRiverBuddies(this.position)
+                .forEach { riverBuddy ->
+                    if (riverBuddy.x == this.position.x && riverBuddy.y == this.position.y + 1) return true
+                }
+        return false
+    }
```

Comment:
also all these copy/pasted code can be unfied as so:

```kotlin

fun hasRiver(xOffset: Int, yOffset: Int): Boolean =
    // disable rivers in the map editor for now. otherwise the game crashes.
    ::tileMap.isInitialized && tileMap.getRiverBuddies(this.position).any { 
         riverBuddy -> riverBuddy.x == this.position.x + xOffset && riverBuddy.y == this.position.y + yOffset 
    }

fun hasBottomRightRiver(): Boolean = hasRiver(-1, 0)
fun hasBottomRiver(): Boolean = hasRiver(-1, -1)
fun hasBottomLeftRiver(): Boolean = hasRiver(0, -1)
fun hasTopLeftRiver(): Boolean = hasRiver(+1, 0)
fun hasTopRiver(): Boolean = hasRiver(+1, +1)
fun hasTopRightRiver(): Boolean = hasRiver(0, +1)

Reply:
Actually, maybe the framework should have a Direction enum that can have some properties which can help simplify a lot of this other code that is based on direction.